src/backend/distributed/operations/repair_shards.c

```diff
@@ -1015,9 +1023,22 @@ EnsureShardCanBeCopied(int64 shardId, const char *sourceNodeName, int32 sourceNo
 																 targetNodePort);
 	if (targetPlacement != NULL)
 	{
-		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-						errmsg("shard " INT64_FORMAT " already exists in the target node",
-							   shardId)));
+		if (targetPlacement->shardState == SHARD_STATE_TO_DELETE)
+		{
+			ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
```

Comment:
Can you add a test that covers this line? (if it's not too hard)

Reply:
I was trying to add a test, it wasn't super easy so I manually tested but it should be possible to add a test for this line so I will try

Reply:
added